src/scrot.c

```diff
@@ -285,33 +287,55 @@ static Imlib_Image scrotGrabAutoselect(void)
     return im;
 }
 
-/* similar to sleep, but deals with EINTR.
- * ideally, we'd want to sleep against an absolute time to prevent any drift
- * but clock_nanosleep doesn't seem to be widely implemented.
- */
-static void scrotSleep(int sec)
+void scrotDoDelay(void)
 {
-    assert(sec > 0);
-    struct timespec delay = { .tv_sec = sec };
-    while (nanosleep(&delay, &delay) < 0 && errno == EINTR);
+    if (!opt.delay)
+        return;
+    if (opt.countdown) {
+        int i;
+
+        dprintf(STDERR_FILENO, "Taking shot in ");
+        for (i = opt.delay; i > 0; i--) {
```

Comment:
I'd declare `i` inside the for loop. We already do it in other places.

```suggestion
        for (int i = opt.delay; i > 0; i--) {
```